### DISCUSSION TITLE
CBC with GPU processing

### DISCUSSION THREAD

**Discussion by IanMoroney** [coin-or/Cbc#675] [Ideas]
I was wondering, considering the fact that CBC only uses CPU resources to calculate and optimise, would it be a good idea to be able to also leverage GPU resources to help with optimisation runs?
If CBC was able to utilise the GPU to run calculations on, in addition or instead of the CPU, i'd imagine it could run a lot quicker.

How feasible would something like this be from a code perspective?

**Comment by jajhall** [comment]
The only competitive optimization solvers that exploit GPUs are the PDLP implementations, but they are a long way off being useful in the context of MIPs. These depend largely on efficient solution of related LPs - for which dual simplex is the only candidate, and it's not going to exploit GPUs any time soon.

**Comment by mgalati13** [comment]
cuOPT (Nvidia) is looking into this. They have successfully done some integration with HiGHs and plan to open-source cuOPT (under COIN).
